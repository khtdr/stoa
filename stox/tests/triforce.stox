#!/usr/bin/env stox

fun triforce() {
  print "triforce:
 ▲
▲ ▲";
  return;
}

triforce();

var i = 0;
var j = 0;

while (true)
  while (i < 7) {
    j = i;
    i = i + 1;
    while (j < 7) {
      j = j + 1;
      if (j == 1) continue;
      if (j == 2) break;
      print "i,j " + i + "," + j;
      if (j == 4) continue 2;
      print "-";
      if (j == 5) break;
      if (j == 6) break 3;
    }
    print "X";
  }

print "done!";
