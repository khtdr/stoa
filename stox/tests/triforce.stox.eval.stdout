triforce:
 ▲
▲ ▲
X
X
i,j 3,3
-
i,j 3,4
i,j 4,4
i,j 5,5
-
X
i,j 6,6
-
done!
